modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -90,14 +91,30 @@ pub struct BitcoinTransactionData {
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
-pub enum DepositState {
+pub enum DepositStateV1 {
     WaitingForTransaction,
     WaitingForConfirmation(BitcoinTransactionData),
     Confirmed(BitcoinTransactionData),
     Claimed(BitcoinTransactionData),
     Failed(String),
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub enum DepositStateV2 {
+    WaitingForTransaction,
+    WaitingForConfirmation {
+        #[serde(with = "bitcoin::amount::serde::as_sat")]
+        btc_deposited: bitcoin::Amount,
+        btc_out_point: bitcoin::OutPoint,
+    },
+    Claimed {
+        #[serde(with = "bitcoin::amount::serde::as_sat")]
+        btc_deposited: bitcoin::Amount,
```

Comment:
@maan2003 and I were looking at this and I learned that this amount does not take into account the fee that we deducted by the federation.. maybe this is OK for now since the fee is a fixed amount.. but probably not the best pattern. I imagine an integrator would want to know the amount of BTC that was actually "claimed" to be used as their e-cash.

Reply:
Yeah, unfortunately it's very hard to accurately show the amount received once e-cash fees are taken into account. So rather than showing a mostly correct value I opted for just supplying the raw deposit value.

This PR is in no way ideal, it's a stop-gap solution till we probably rebuild the pegin monitor to allow for better observeability.